arangod/Utils/SupportInfoBuilder.cpp

```diff
@@ -0,0 +1,762 @@
+      options.param("support", "true");
+
+      size_t coordinators = 0;
+      size_t dbServers = 0;
+      size_t agents = 0;
+
+      std::vector<std::string> dbServerNames;
+      ClusterInfo& ci = server.getFeature<ClusterFeature>().clusterInfo();
+      for (auto const& server : ci.getServers()) {
+        if (server.first.starts_with("CRDN")) {
+          ++coordinators;
+        } else if (server.first.starts_with("PRMR")) {
+          dbServerNames.emplace_back(server.first);
+          ++dbServers;
+        } else if (server.first.starts_with("SNGL")) {
+          // SNGL counts as DB server here
+          TRI_ASSERT(isActiveFailover);
+          ++dbServers;
+        }
+        if (server.first == ServerState::instance()->getId()) {
+          // ourselves!
+          continue;
+        }
+
+        std::string reqUrl =
+            isTelemetricsReq ? "/_admin/telemetrics" : "/_admin/support-info";
+        auto f = network::sendRequestRetry(
+            pool, "server:" + server.first, fuerte::RestVerb::Get, reqUrl,
+            VPackBuffer<uint8_t>{}, options, ::buildHeaders());
+        futures.emplace_back(std::move(f));
+      }
+
+      VPackBuilder dbInfoBuilder;
+      if (!futures.empty()) {
+        dbInfoBuilder.openObject();
+        auto responses = futures::collectAll(futures).get();
+        for (auto const& it : responses) {
+          auto& resp = it.get();
+          auto res = resp.combinedResult();
+          if (res.fail()) {
+            LOG_TOPIC("4800b", WARN, Logger::STATISTICS)
+                << "Failed to get server info: " << res.errorMessage();
+          } else {
+            auto slice = resp.slice();
+            // copy results from other server
+            if (slice.isObject()) {
+              std::string hostId =
+                  basics::StringUtils::replace(resp.destination, "server:", "");
+              if (isTelemetricsReq) {
+                result.add(slice.get("host"));
+              } else {
+                result.add(hostId, slice.get("host"));
+              }
+              if (isTelemetricsReq) {
+                auto const databasesSlice = slice.get("databases");
+                if (!databasesSlice.isNone()) {
+                  dbInfoBuilder.add(hostId, databasesSlice);
+                }
+              }
+            }
+          }
+        }
+        dbInfoBuilder.close();
+      }
+
+      result.close();  // servers
+
+      auto manager = AsyncAgencyCommManager::INSTANCE.get();
+      if (manager != nullptr) {
+        agents = manager->endpoints().size();
+      }
+
+      result.add("agents", VPackValue(agents));
+      result.add("coordinators", VPackValue(coordinators));
+      std::string dbServersKey = "dbServers";
+      std::string shardsStatitsics = "shards";
+      if (isTelemetricsReq) {
+        dbServersKey = "db_servers";
+        shardsStatitsics = "shards_statistics";
+      }
+      result.add(dbServersKey, VPackValue(dbServers));
```

Comment:
This mix of snake_case and camelCase looks really weird.

Reply:
As agreed, I used the same builder structure for API `/_admin/support-info` and `/_admin/telemetrics`, so I have to make this distinction

Reply:
To be able to gather the telemetrics response following its rules and maintaining `support-info` as it is, with no changes. In the former comment, `persistedId` was actually wrong and I changed it to `persisted_id`.